keytree/src/lib.rs

```diff
@@ -139,6 +155,22 @@ impl Xpub {
         }
     }
 
+    /// Returns a leaf Xpub, which can safely be shared. (Is this really an xprv, or just a prv?)
+    /// Users must provide customize, in order to separate sibling keys from one another
+    /// through unique derivation paths.
+    pub fn derive_leaf_key(&self, customize: impl FnOnce(&mut Transcript)) -> CompressedRistretto {
+        let mut t = Transcript::new(b"Keytree.derivation");
+        t.commit_bytes(b"pt", self.precompressed_pubkey.as_bytes());
+        t.commit_bytes(b"dk", &self.dk);
+
+        // change the derivation path for this key
+        customize(&mut t);
+
+        // squeeze a challenge scalar
+        let f = t.challenge_scalar(b"f.leaf");
+        (self.point + (f * &constants::RISTRETTO_BASEPOINT_POINT)).compress()
```

Comment:
maybe add a comment about future plans?
```suggestion
        // TBD: change this to VerificationKey after we factor out Musig (#239) 
        //      in order to preserve uncompressed point.
        (self.point + (f * &constants::RISTRETTO_BASEPOINT_POINT)).compress()
```